### PR TITLE
Make 'setUploadConsignmentStatusValueToComplete' more generic

### PR DESCRIPTION
**Pull request by techncl**
...by making it an 'updateConsignmentStatus' method.

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentStatusFields.scala [hunk 1]
```diff
@@ -16,7 +16,11 @@ object ConsignmentStatusFields {
   val mutationFields: List[Field[ConsignmentApiContext, Unit]] = fields[ConsignmentApiContext, Unit](
     Field("markUploadAsCompleted", OptionType(IntType),
       arguments = ConsignmentIdArg :: Nil,
-      resolve = ctx => ctx.ctx.consignmentStatusService.setUploadConsignmentStatusValueToComplete(ctx.arg(ConsignmentIdArg)),
+      resolve = ctx => ctx.ctx.consignmentStatusService.updateConsignmentStatus(
+        ctx.arg(ConsignmentIdArg),
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentStatusFields.scala:20]: Hardcoding these values should be a temporary measure. We should:
1. create new generic mutation: `updateConsignmentStatus` which takes the relevant arguments
2. commit these changes
3. move any clients using the old mutation: `markUploadAsCompleted`
4. remove the old mutation

**Review by TomJKing — COMMENTED** [review]
Looks good. I've added a comment about creating a new generic mutation. Thanks

**Review by TomJKing — COMMENTED** [review]
Looks good. One thing that has occurred to me, is that we should probably be checking the `statusType` and `statusValue` inputs are valid.

Potentially create a companion object in the `ConsignmentStatusService.scala` and do something like this:

```
object ConsignmentStatusService {
  val exportStatusType = "Export"
  val exportStatusValues = Set("Upload", "Completed")

  val consignmentStatuses: Map[String, Set[String]] = Map(exportStatusType -> exportStatusValues)

  def validateConsignmentStatus(statusType: String, statusValue: String): Boolean = {
    if(consignmentStatuses.contains(statusType)) {
      consignmentStatuses.get(statusType).fold(
        throw InputDataException(s"Invalid consignment status value '$statusValue' for status type '$statusType'"))(_.contains(statusValue))
    } else {
      throw InputDataException(s"Invalid consignment status type '$statusType'")
    }
  }
}
```
Sorry, only thought of this now.

File: .bsp/sbt.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"name":"sbt","version":"1.6.2","bspVersion":"2.0.0-M5","languages":["scala"],"argv":["/Library/Java/JavaVirtualMachines/jdk-15.0.2.jdk/Contents/Home/bin/java","-Xms100m","-Xmx100m","-classpath","/Users/nlutale/Library/Application Support/JetBrains/IntelliJIdea2021.3/plugins/Scala/launcher/sbt-launch.jar","xsbt.boot.Boot","-bsp","--sbt-launch-jar=/Users/nlutale/Library/Application%20Support/JetBrains/IntelliJIdea2021.3/plugins/Scala/launcher/sbt-launch.jar"]}
```
Comment (TomJKing) [inline .bsp/sbt.json:1]: This file shouldn't be merged. Should probably be added to the `.gitignore` file

**Review by TomJKing — COMMENTED** [review]
Looks good. Just one minor comment.

File: .bsp/sbt.json (hunk 2 above)
Comment (techncl) [reply .bsp/sbt.json:1]: 🤦‍♂️ , I did a `git add .`

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**